### ISSUE TITLE
`@stylistic/string-quotes` leaves a value carrying an inline comment half-checked under `postcss-scss`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#100]
Related to #31, and the half of stylelint/stylelint#6314 its report calls out with “(with `//` the same offenses)”.

Found while fixing #33 and #61. `postcss-scss` rewrites every `//` comment inside `raws[prop].raw` into `/* … */`, keeps the source spelling in a third field, `raws[prop].scss`, and its stringifier prefers that field over the raw (`lib/scss-stringifier.js:44`). The rule reads and writes the raw, so for a value carrying such a comment both halves fail at once: a position behind the first rewritten comment drifts two characters per comment, and a fix written into the raw reaches no output.

The fix for #33 and #61 therefore fenced the case off rather than delivering it wrong. Two gaps are left behind, and the `Unreleased` section of the changelog records them as such.

### What minimal example or steps are needed to reproduce the bug?

A string behind an inline comment:

```scss
$m: (
  // c
  'a': 1
);
```

A string in front of one:

```scss
$m: (
  'a': 1 // c
);
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/string-quotes": "double"
  }
}
```

### What did you expect to happen?

Both values are reported — at `3:3` and at `2:3` — and `--fix` rewrites the quotation marks of both, leaving the comments exactly as they are.

### What actually happened?

The first is not reported at all. The second is reported, but `--fix` declines it and the warning stands; the code is at least left intact.

### Do you have a proposal to fix the bug?

Read and write `raws[prop].scss` when it is there, instead of only `.raw`. That field is the text of the file, with the `//` spelling intact, so the positions come out right, and it is the field the stringifier prints, so a fix written into it arrives. The same repair was already made for selectors in `selectorCombinatorSpaceChecker` (`0ff6dae`).

This had to wait for #32, now fixed. With `//` comments back in the scanned text, the value tokenizer takes the quotation marks inside them for code again — `$m: ('a': 1 // Some "comment")` yields a string node at index 19 of `.scss` — and the span scan added for #32 is what passes over it. Lifting the fence any earlier would only have moved that defect from Less into SCSS.

Two things go with the change:

- the `accept` case covering the first example in `lib/rules/string-quotes/index.test.js` becomes a `reject`, since `'a'` is a genuine violation the fence currently swallows — it is the only test that fails when the fence is disabled;
- the changelog bullet describing the fence is deleted outright rather than superseded, since it sits in `Unreleased` and no release has ever carried it.

Worth doing before `getAtRuleParams` and `setAtRuleParams` gain their second caller in the `indentation` work for #64 and #65, so that work is written against their final shape.
